Graduate `UseStrimziPodSets` feature gate and remove `StatefulSet` support (#8273)
strimzi/strimzi-kafka-operator: documentation/modules/deploying/con-deploy-custom-resources-example.adoc

The Kubernetes API server allows custom resources to be created based on the `kind` and understands from the CRD how to validate and store the custom resource when it is added to the Kubernetes cluster.

> WARNING: When CRDs are deleted, custom resources of that type are also deleted. Additionally, the resources created by the custom resource, such as pods and services are also deleted.

Reviewer: ```suggestion
WARNING: When a `CustomResourceDefinition` is deleted, custom resources of that type are also deleted. Additionally, Kubernetes resources created by the custom resource are also deleted, such as `Deployment`, `Pod`, `Service` and `ConfigMap` resources.
```
